### PR TITLE
enable dependabot for dependency updates

### PR DESCRIPTION
**Pull request by xvrl**
As discussed in the dev@ mailing list I am proposing we start using GitHub's dependabot to manage updates.
If we agree, then checking in this file should automatically enable dependabot for us.

### CODE REVIEW

**Comment by suneet-s** [comment]
I'm +1 for on this approach. I think our IT coverage is quite good at this point (except maybe for Hadoop).

Could we customize the message in the PR that this bot creates to warn reviewers of any gotcha's like kinesis tests not running as part of CI, missing hadoop tests, etc.

File: .github/dependabot.yml
Comment (asdf2014) [inline .github/dependabot.yml:7]: I think we should add the `open-pull-requests-limit: 10` configuration to limit the number of open PRs for easy access and maintenance.

File: .github/dependabot.yml
Comment (asdf2014) [inline .github/dependabot.yml:6]: It would be better to add the `time: "21:00"` option under `schedule`, and then we can maintain dependency upgrades at a fixed time every day, which is more convenient than random time.

**Review by asdf2014 — APPROVED** [review]
Hi, @xvrl . Thank you to bring this up. As far as I know, the `Dependabot Preview` will be shut down on August 3rd, 2021. In order to keep getting `Dependabot` updates, we must merge this PR and migrate to GitHub-native `Dependabot` before then. Overall LGTM, I also left some comments for you, please check it out.

**Comment by xvrl** [comment]
> As far as I know, the Dependabot Preview will be shut down on August 3rd, 2021. In order to keep getting Dependabot updates, we must merge this PR and migrate to GitHub-native Dependabot before then

@asdf2014 my PR is already for the native github integration, since it is part of the `.github` directory, as decribed here https://docs.github.com/en/code-security/supply-chain-security/enabling-and-disabling-version-updates#about-version-updates-for-dependencies

File: .github/dependabot.yml
Comment (xvrl) [reply .github/dependabot.yml:6]: I don't think we need to change this, since the default is already fixed at `5:00 UTC` according to https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates#scheduletime

File: .github/dependabot.yml
Comment (xvrl) [reply .github/dependabot.yml:7]: the default is already limited to 5 PRs, so I think we are good (see https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates#open-pull-requests-limit)

Comment (asdf2014) [reply .github/dependabot.yml:7]: Good to know, thanks :+1:

File: .github/dependabot.yml
Comment (asdf2014) [reply .github/dependabot.yml:6]: Cool :+1:

**Comment by jihoonson** [comment]
Hmm I'm a bit worried about this change. There are many integration tests that don't run on Travis because they require extra setups such as s3 credentials and a bucket. Should we resolve this issue first? I also think it would be better to move hadoop out as a separate extension from core before we do this.

**Comment by xvrl** [comment]
> There are many integration tests that don't run on Travis because they require extra setups such as s3 credentials and a bucket. Should we resolve this issue first?

@jihoonson if we don't have a good way of catching these things today as part of our CI builds, I'm not sure there is a difference between automation or humans submitting PRs. We still need to approve these changes before they get merged, so I would hope that we don't approve upgrades if we are aware that it might affect some of those tests.

We can of course exclude any dependencies that we know lack good integration test coverage to be safe.

**Review by gianm — APPROVED** [review]
LGTM, but IMO especially at first we need to make sure to avoid applying these updates without testing. Let's apply the ones that we are confident in our ability to test, and add ignores (or tests!) for the rest. I wrote back to the dev thread suggesting the same.

**Comment by jihoonson** [comment]
> @jihoonson if we don't have a good way of catching these things today as part of our CI builds, I'm not sure there is a difference between automation or humans submitting PRs. We still need to approve these changes before they get merged, so I would hope that we don't approve upgrades if we are aware that it might affect some of those tests.
> 
> We can of course exclude any dependencies that we know lack good integration test coverage to be safe.

The difference will be that the PR author will be responsible for testing in human-oriented way (that is, the testing burden is more distributed), while it will be reviewers in the automatic way until we will have some automatic testing. This will increase the burden of the reviewers, especially given that we don't have many active reviewers today. But I agree that having more tests running on Travis solves the root cause.

I'm wondering if we can have an allow list for dependencies that we are confident to upgrade automatically instead of using a deny list. Assuming the allow list will contain only the dependencies we know their testing scope, I would be more confident to merge the PRs that this bot will create without actually testing them by myself.

**Pull request merged**
